File: app/src/main/java/com/magnitudestudios/GameFace/Bases/BasePermissionsActivity.java
package com.magnitudestudios.GameFace.Bases;

import android.content.Context;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;

import static com.magnitudestudios.GameFace.Constants.ALL_PERMISSIONS;
import static com.magnitudestudios.GameFace.Constants.PERMISSIONS;

public class BasePermissionsActivity extends AppCompatActivity {

    private static final String TAG = "BasePermissionsActivity";
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        checkPermissions(ALL_PERMISSIONS);

    }

    public static boolean hasPermissions(Context context, String[] permissions) {
        if (context != null && permissions != null) {
            for (String permission : permissions) {
                if (ActivityCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
                    return false;
                }
            }
        }
        return true;
    }

    // Function to check and request permission.
    public void checkPermissions(int requestCode)
    {
        if (!hasPermissions(this, PERMISSIONS)) {
            ActivityCompat.requestPermissions(this, PERMISSIONS, requestCode);
        }
    }

    public boolean hasAllPermissionsGranted(@NonNull int[] grantResults) {
        for (int grantResult : grantResults) {
            if (grantResult == PackageManager.PERMISSION_DENIED) {
                return false;
            }
        }
        return true;
    }


    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults)
    {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);

        Log.d(TAG, "onRequestPermissionsResult: "+ permissions.toString() + grantResults.toString());

        if (requestCode == ALL_PERMISSIONS && grantResults.length > 0) {
            if (hasAllPermissionsGranted(grantResults)) {
                Toast.makeText(this, "All Permissions Granted", Toast.LENGTH_SHORT).show();
            }
            else {
                Toast.makeText(this, "All Permission Not Granted", Toast.LENGTH_SHORT).show();
            }
        }

    }
}

File: app/src/main/java/com/magnitudestudios/GameFace/Bases/BasePermissionsActivity.kt
package com.magnitudestudios.GameFace.Bases

import android.annotation.SuppressLint
import android.content.Context
import android.content.pm.PackageManager
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.core.app.ActivityCompat
import com.magnitudestudios.GameFace.Constants

@SuppressLint("Registered")
open class BasePermissionsActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        checkPermissions(Constants.ALL_PERMISSIONS)
    }

    // Function to check and request permission.
    fun checkPermissions(requestCode: Int) {
        if (!hasPermissions(this, Constants.PERMISSIONS)) {
            ActivityCompat.requestPermissions(this, Constants.PERMISSIONS, requestCode)
        }
    }

    fun hasAllPermissionsGranted(grantResults: IntArray): Boolean {
        for (grantResult in grantResults) {
            if (grantResult == PackageManager.PERMISSION_DENIED) {
                return false
            }
        }
        return true
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        Log.d(TAG, "onRequestPermissionsResult: $permissions$grantResults")
        if (requestCode == Constants.ALL_PERMISSIONS && grantResults.size > 0) {
            if (hasAllPermissionsGranted(grantResults)) {
                Toast.makeText(this, "All Permissions Granted", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this, "All Permission Not Granted", Toast.LENGTH_SHORT).show()
            }
        }
    }

    companion object {
        private const val TAG = "BasePermissionsActivity"
        fun hasPermissions(context: Context?, permissions: Array<String>): Boolean {
            if (context != null) {
                for (permission in permissions) {
                    if (ActivityCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
                        return false
                    }
                }
            }
            return true
        }
    }
}